examples/protobuf_parser.py

```diff
@@ -27,8 +27,8 @@
                     fixed32 fixed64 sfixed32 sfixed64 bool string bytes""") | ident
 rvalue = integer | TRUE_ | FALSE_ | ident
 fieldDirective = LBRACK + Group(ident + EQ + rvalue) + RBRACK
-fieldDefn = (( REQUIRED_ | OPTIONAL_ | REPEATED_ )("fieldQualifier") -
-              typespec("typespec") + ident("ident") + EQ + integer("fieldint") + ZeroOrMore(fieldDirective) + SEMI)
+fieldDefnPrefix =  REQUIRED_ | OPTIONAL_ | REPEATED_
+fieldDefn = (Optional(fieldDefnPrefix) + typespec("typespec") + ident("ident") + EQ + integer("fieldint") + ZeroOrMore(fieldDirective) + SEMI)
```

Comment:
Please restore "fieldQualifier" as a results name on `fieldDefnPrefix`, something like:

    fieldDefn = (Optional(fieldDefnPrefix("fieldQualifier")) + typespec("typespec") + etc.

Reply:
Looking at the spec further, this might be better named "fieldRule". In proto3, I don't see "required" or "optional", but there are references to "singular" and "repeated". Is there an actual BNF somewhere?

Reply:
As I know "required" are optional, but still in syntax.